tests/restapi/test_threaded_restapi.py

```diff
@@ -0,0 +1,140 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=protected-access
+"""Tests for the `aiida.restapi` module, using it in threaded mode.
+
+Threaded mode is the default (and only) way to run the AiiDA REST API (see `aiida.restapi.run_api:run_api()`).
+This test file's layout is inspired by https://gist.github.com/prschmid/4643738
+"""
+from threading import Thread
+
+import requests
+from werkzeug.serving import make_server
+import pytest
+
+from aiida.restapi.common.config import CLI_DEFAULTS, API_CONFIG
+from aiida.restapi.run_api import configure_api
+
+# The URL to the REST API we will start up.
+# This is the default location where Flask apps start when using the built in web server.
+SERVER_URL = 'http://{hostname}:{port}{api}'.format(
+    hostname=CLI_DEFAULTS['HOST_NAME'], port=CLI_DEFAULTS['PORT'], api=API_CONFIG['PREFIX']
+)
+
+NO_OF_REQUESTS = 100
+
```

Comment:
These tests look good, there is just another opportunity to leverage pytest's amazing fixture system. You could turn the threaded REST API server into a fixture and reuse it. Something like
```
@pytest_fixture
def restapi_server():
    from aiida.restapi.common.config import CLI_DEFAULTS, API_CONFIG

    server_url = 'http://{hostname}:{port}{api}'.format(
        hostname=CLI_DEFAULTS['HOST_NAME'], port=CLI_DEFAULTS['PORT'], api=API_CONFIG['PREFIX']
    )
    aiida_restapi = configure_api()
    server = make_server(
        host=CLI_DEFAULTS['HOST_NAME'],
        port=int(CLI_DEFAULTS['PORT']),
        app=aiida_restapi.app,
        threaded=True,
        processes=1,
        request_handler=None,
        passthrough_errors=False,
        ssl_context=None,
        fd=None
    )

    server_thread = Thread(target=server.serve_forever)

    try:
        server_thread.start()
        yield server_url
    except Exception as exc:  # pylint: disable=broad-except
        pytest.fail('Something went horribly wrong! {}'.format(repr(exc)))
    finally:
        server.shutdown()


def test_run_threaded_server(aiida_localhost, restapi_server):
    server_url = restapi_server

    for _ in range(NO_OF_REQUESTS):
        response = requests.get(server_url + '/computers/{}'.format(computer_id))

        assert response.status_code == 200

        try:
            response_json = response.json()
        except ValueError:
            pytest.fail('Could not turn response into JSON. Response: {}'.format(response.raw))
        else:
            assert 'data' in response_json
```

You can then reuse the same fixture in the second test

Reply:
This was the incentive of making it a class originally ...
And yeah, there are definitely improvements to be done here, and this seems a good one.
For now, I simply want the tests to work with the CI. They work fine locally ...